orchagent/fgnhgorch.cpp

```diff
@@ -1640,6 +1940,19 @@ bool FgNhgOrch::doTaskFgNhgPrefix(const KeyOpFieldsValuesTuple & t)
             return false;
         }
 
+        // In the dynamic_nhg mode, only 1 FG_NHG_PREFIX is allowed for each FG_NHG. If ip_prefix 
+        // doesn't match with the existing prefixes in the fgNhgEntry, then return false.
+        if (fgNhg_entry->second.nhg_mode == DYNAMIC_FGNHG)
+        {
+            if (fgNhg_entry->second.prefixes.size() != 0)
```

Comment:
why is it the case that there can only be 1 prefix per FG_NHG? I think allowing multiple prefixes is possible associated with the same set of FG_NHG config. FG_NHG is intended to be shared config across a bunch of prefixes/next-hop group definitions.

Reply:
In this case, list of nexthop IPs is not statically configured, and it's quite possible that routing protocol will push a different set of nexthop IPs for each of the prefixes and we don't have a mechanism to create a common group out of them. So, we have an explicit check to avoid that configuration.

Reply:
The following is just a suggestion and more for future/a follow up PR if there is ever a need for it: there are 2 different structures in the code, FGNextHopGroupEntry for per prefix things via BGP/route entry, and FgNhgEntry for user defined fine grained nhg config.
 
Given the new use case where you need nexthops/links from the route entry instead of user config, my suggestion is to try adding a new pointer to a nexthops and links struct and storing it in the FGNextHopGroupEntry, and then refactoring to use nexthops/links from that route entry vs as the code is currently doing things from FgNhgEntry. That way you can support multiple prefixes per FG_NHG(which sounds like the ideal design imho).